### PR TITLE
Add --version flag as alias for version command

### PR DESCRIPTION
**Pull request by osterman**
## what
- Add `--version` persistent flag to RootCmd as a simple alias for `atmos version` command
- Update `isVersionCommand()` helper to detect both `version` and `--version` forms
- Add test case for `atmos --version` in empty directory (no config required)
- Add test case for `--version` flag to `TestIsVersionCommand`

## why
- Provides a conventional CLI experience - most command-line tools support `--version` flag
- Aligns with standard POSIX/GNU conventions for version flags
- Makes version checking easier for users and scripts
- Both `atmos version` and `atmos --version` now work without requiring `atmos.yaml` configuration

## references
- `atmos --version` outputs simple format: `atmos <version>`
- `atmos version` outputs full styled format with update checks
- The `--version` flag does not support additional flags like `--check` or `--format` (use `atmos version` for those features)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/terraform_generate_backend_test.go [hunk 1]
```diff
@@ -21,6 +22,11 @@ func TestTerraformGenerateBackendCmd(t *testing.T) {
 	err = os.Setenv("ATMOS_LOGS_LEVEL", "Debug")
 	assert.NoError(t, err, "Setting 'ATMOS_LOGS_LEVEL' environment variable should execute without error")
 
+	// Reset flag states to prevent pollution from other tests
+	RootCmd.Flags().Visit(func(f *pflag.Flag) {
+		f.Changed = false
+	})
+
```
[the remark this answers is not in the document]
Comment (osterman) [reply cmd/terraform_generate_backend_test.go]: Thank you for the review! The current implementation deliberately uses `Visit()` instead of `VisitAll()` for the following reasons:

1. **It already addresses your concern** - The code calls `f.Value.Set(f.DefValue)` before setting `f.Changed = false`, so both the value AND the Changed bit are being reset properly.

2. **Avoids issues with complex flag types** - During testing, we found that `VisitAll()` caused issues with certain complex flag types. Using `Visit()` only resets flags that were actually changed, avoiding potential side effects from resetting flags that were never modified.

3. **More targeted approach** - By only iterating over changed flags, we avoid touching flags that are in their default state, which is safer for flags with complex initialization logic.

The commit message explicitly mentions this decision: "Only reset flags that were actually changed to avoid issues with complex flag types."

All tests pass with this approach, and it provides better isolation between tests without the risks associated with `VisitAll()`.
